### PR TITLE
readme update

### PR DESCRIPTION
**Pull request by jandjorgensen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+This algorithm was originally discussed in [this StackExchange thread](http://cs.stackexchange.com/questions/43447/order-preserving-update-of-a-sublist-of-a-list-of-mutable-objects-in-sublinear-t/44502#44502).
+
+### When to Use
+
+In general, it is preferable to use `can-derive` over alternative approaches when:
+
+- Your source list contains 10 or more items
+- You would need to perform a `diff` operation in an alternative approach
+
+The latter case will be fulfilled when you need to render the filtered list in the DOM. Since `can-derive` generates change events, the `diff` step is no longer necessary when re-rendering a filtered list in the DOM.
```
Comment (akagomez) [inline README.md]: > the `diff` step is no longer necessary when re-rendering a filtered list in the DOM.

I think this should be "no longer necessary when updating the DOM representation of the filtered list". 

Or maybe there's a better way to say it. I specifically don't like the "re-rendering" bit.

File: README.md [hunk 2]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+- Listening to when the result of the predicate function changes for any item
+
+This algorithm was originally discussed in [this StackExchange thread](http://cs.stackexchange.com/questions/43447/order-preserving-update-of-a-sublist-of-a-list-of-mutable-objects-in-sublinear-t/44502#44502).
+
+### When to Use
+
+In general, it is preferable to use `can-derive` over alternative approaches when:
+
+- Your source list contains 10 or more items
+- You would need to perform a `diff` operation in an alternative approach
```
Comment (akagomez) [inline README.md]: I don't think we should assume that the end-user knows a `diff` is occurring to qualify the use of can-derive. 

- You need to know how the filtered list changed, for instance when rendering in the DOM (which typically depends on a `diff`).

File: README.md [hunk 3]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+
+- push
+- pop
+- shift
+- unshift
+- splice
+
+## Performance
+
+`can-derive` optimizes for indertions and removals, completing them in O(log n) time. This means that changes to the source list will automatically update the derived list in O(log n) time, compared to the standard O(n) time you would expect in other implementations.
```
Comment (akagomez) [inline README.md]: Typo: "indertions" > "insertions"

Also, I think the "O(log n)" bits should be code formatted like, `O(log n)`. But I may be wrong about that.

File: README.md [hunk 4]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+
+- [filter](http://canjs.com/docs/can.List.prototype.filter.html)
+- [indexOf](http://canjs.com/docs/can.List.prototype.indexOf.html)
+- [map](http://canjs.com/docs/can.List.prototype.map.html)
+- [replace](http://canjs.com/docs/can.List.prototype.replace.html)
+- [slice](http://canjs.com/docs/can.List.prototype.slice.html)
+
+### Disabled can.List methods
+
+The filtered list is not changed manually, but is maintained as the source list changes. Because of this, the following `can.List` methods are disabled:
```
Comment (akagomez) [inline README.md]: I forgot all about this. Well said.

File: README.md [hunk 5]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+
+### each
+
+`derivedList.each(fn) --> DerivedList`
+
+Iterates through the DerivedList, calling a function for each element. The elements provided inside the function will be the elements *in the source list*. The filtered list will be updated only if the outcome of the predicate function changes.
+
+### Other can.List methods
+
+Since DerivedList is a [can.List](http://canjs.com/docs/can.List.html), the following methods are also available:
```
Comment (akagomez) [inline README.md]: DerivedList isn't a `can.List` exactly. It's a `can.RBTreeList`, which is meant to be `can.List` compliant.

File: README.md [hunk 6]
```diff
@@ -17,13 +37,112 @@ var completed = sourceList.filter(function(todo){
...
+
+`derivedList.attr(index) --> Object`
+
+Reads an element from an index on the filtered list. This references the *element in the source list* that *is at the specified index in the filtered list*.
+
+### each
+
+`derivedList.each(fn) --> DerivedList`
+
+Iterates through the DerivedList, calling a function for each element. The elements provided inside the function will be the elements *in the source list*. The filtered list will be updated only if the outcome of the predicate function changes.
```
Comment (akagomez) [inline README.md]: > The filtered list will be updated only if the outcome of the predicate function changes.

This seems out of place.

File: README.md (hunk 5 above)
Comment (jandjorgensen) [inline README.md]: "DerivedList inherits from `can.List`"

File: README.md (hunk 6 above)
Comment (jandjorgensen) [inline README.md]: "Right message, wrong place"

**Pull request merged**